src/geocel/detail/PolygonUtils.hh

```diff
@@ -0,0 +1,126 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/detail/LengthUnits.hh
+//! \brief NOTE: only use inside geocel; prefer celeritas/Units.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+//#define CELER_ICRT inline constexpr real_type
+#include <iostream>
+#include <vector>
+#include <corecel/cont/Array.hh>
+#include <corecel/math/ArrayUtils.hh>
+#include <orange/OrangeTypes.hh>
+
+namespace celeritas
+{
+namespace geoutils
+{
```

Comment:
There's not yet a `geoutils` namespace, so just put this in `celeritas`. This file should not be in a `detail` directory if it lives in `geocel/` and is used by `orange/orangeinp`. I think `geocel` is a fine place for it since it's generic, but I can't see these being used outside of orange—so perhaps `orange/orangeinp/detail` in the `celeritas::orangeinp::detail` namespace is better?